importer/tests/test_views.py

```diff
@@ -30,7 +33,53 @@ def test_upload_get(self):
     def test_upload_get_bad_repo(self):
         """GET upload page."""
         resp = self.client.get("/repositories/999/import", follow=True)
+        self.assertTrue(resp.status_code == NOT_FOUND)
+
+    def test_upload_with_permissions(self):
+        """GET upload page with different user permissions"""
+        self.logout()
+        self.login(self.USERNAME_NO_REPO)
+        # user has no permissions at all
+        resp = self.client.get("/repositories/{0}/import".format(
+            self.repo.slug), follow=True)
         self.assertTrue(resp.status_code == UNAUTHORIZED)
+        # user has author permissions and cannot see the import for the repo
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_author
+        )
+        resp = self.client.get("/repositories/{0}/import".format(
+            self.repo.slug), follow=True)
+        self.assertTrue(resp.status_code == UNAUTHORIZED)
```

Comment:
I know this is a test of imports and not the repository listing, but it would add clarity to test for a 403 on the repo view page, then test for a 200 (but still a 403 on the import), just to show that the permissions are working properly.